packages/nouns-contracts/contracts/StreamEscrow.sol

```diff
@@ -0,0 +1,307 @@
+    mapping(uint256 streamId => Stream) internal streams;
+    mapping(address => bool) public allowedToCreateStream;
+
+    /**
+     * @notice Initializes the StreamEscrow contract
+     * @param daoExecutor_ The address of the DAO executor contract
+     * @param ethRecipient_ The address that will receive ETH payments
+     * @param nounsRecipient_ The address that will receive Nouns tokens when streams are canceled
+     * @param nounsToken_ The address of the Nouns ERC721 token contract
+     * @param streamCreator_ The address that will be initially allowed to create streams
+     */
+    constructor(
+        address daoExecutor_,
+        address ethRecipient_,
+        address nounsRecipient_,
+        address nounsToken_,
+        address streamCreator_
+    ) {
+        daoExecutor = daoExecutor_;
+        ethRecipient = ethRecipient_;
+        nounsRecipient = nounsRecipient_;
+        nounsToken = INounsToken(nounsToken_);
+        allowedToCreateStream[streamCreator_] = true;
+    }
+
+    /**
+     * @notice Forwards all streams and creates a new stream for a Noun.
+     * @notice ETH value must be sent with this function call.
+     * @dev Combines forwardAll() and createStream() operations into a single transaction.
+     * @param nounId The ID of the Noun token for which the stream is being created.
+     * @param streamLengthInTicks The duration of the stream in ticks. 1 day must pass for a tick to increase.
+     */
+    function forwardAllAndCreateStream(uint256 nounId, uint16 streamLengthInTicks) external payable {
+        forwardAll();
+
+        createStream(nounId, streamLengthInTicks);
+    }
+
+    /**
+     * @notice Creates a new ETH stream for a specific Noun token.
+     * @dev Only allowed addresses that are also Noun owners/approved operators can create streams.
+     * @param nounId The ID of the Noun token to create a stream for.
+     * @param streamLengthInTicks The duration of the stream in ticks.
+     * @custom:throws 'not allowed' if sender is not allowed to create streams.
+     * @custom:throws 'only noun owner or approved' if sender is not owner/approved.
+     * @custom:throws 'stream active' if a stream already exists for the Noun.
+     * @notice ETH value must be sent with this function call.
+     */
+    function createStream(uint256 nounId, uint16 streamLengthInTicks) public payable {
+        require(allowedToCreateStream[msg.sender], 'not allowed');
+        require(isApprovedOrOwner(msg.sender, nounId), 'only noun owner or approved');
+        require(!isStreamActive(nounId), 'stream active');
+
+        // register new stream
+        uint128 ethPerTick = toUint128(msg.value / streamLengthInTicks);
+        uint32 streamLastTick = currentTick + streamLengthInTicks; // streamLastTick is inclusive
+        ethStreamEndingAtTick[streamLastTick] += ethPerTick;
+
+        // the remainder is immediately streamed to the DAO
+        uint256 remainder = msg.value % streamLengthInTicks;
```

Comment:
this would feel safer if instead of modulo it was `msg.value - (ethPerTick * streamLengthInTicks)`. haven't given deep thought to whether we might have miscalculations due to reducing bits or anything like that, just think this reflects more accurately the math of how the stream will operate, leaving less room for problems, and makes it easier to read with ease.

wdyt?